java/gui/src/main/java/com/squareup/subzero/wallet/WalletLoader.java

```diff
@@ -23,13 +23,16 @@
  * file.
  */
 public class WalletLoader {
-  private final static String HD_PATH = "/data/app/subzero/wallets/";
   protected final static String MARKER_FILE = ".subzero_702e63a9";
 
   private Path directory;
 
-  public WalletLoader() {
-    directory =  FileSystems.getDefault().getPath(HD_PATH);
+  /**
+   * Creates a new WalletLoader and sets the wallet file location to the path provided.
+   * Will throw runtime exceptions if the path provided is invalid.
+   */
+  public WalletLoader(String walletFilePath) {
+    directory =  FileSystems.getDefault().getPath(walletFilePath);
```

Comment:
super nit: this can call the other constructor instead: `this(FileSystems ...)`, this way if the Path constructor gets more logic later we don't need to duplicate it here.